internal/node/model/models.go

```diff
@@ -46,39 +41,60 @@ const (
 	DefaultBlockStatusSafe      DefaultBlock = "SAFE"
 )
 
+const (
+	EpochStatusReceivingInputs    EpochStatus = "RECEIVING_INPUTS"
+	EpochStatusReceivedLastInput  EpochStatus = "RECEIVED_LAST_INPUT"
+	EpochStatusProcessedAllInputs EpochStatus = "PROCESSED_ALL_INPUTS"
+	EpochStatusCalculatedClaim    EpochStatus = "CALCULATED_CLAIM"
+	EpochStatusSubmittedClaim     EpochStatus = "SUBMITTED_CLAIM"
+	EpochStatusAcceptedClaim      EpochStatus = "ACCEPTED_CLAIM"
+	EpochStatusRejectedClaim      EpochStatus = "REJECTED_CLAIM"
+)
+
 type NodePersistentConfig struct {
 	DefaultBlock            DefaultBlock
 	InputBoxDeploymentBlock uint64
 	InputBoxAddress         Address
 	ChainId                 uint64
 	IConsensusAddress       Address
+	EpochLength             uint64
 }
 
 type Application struct {
 	Id                 uint64
 	ContractAddress    Address
 	TemplateHash       Hash
-	SnapshotURI        string
 	LastProcessedBlock uint64
-	EpochLength        uint64
 	Status             ApplicationStatus
 }
 
+type Epoch struct {
+	Id              uint64
+	AppAddress      Address
+	Index           uint64
+	FirstBlock      uint64
+	LastBlock       uint64
+	ClaimHash       *Hash
+	TransactionHash *Hash
+	Status          EpochStatus
+}
+
```

Comment:
For formatting sake, let's put the foreign keys as the last values on the struct